### PR TITLE
RouteMatrix 2024-06-01-preview

### LINKED ISSUE #28960: [Azure Maps - Azure Maps] API Review
New API Review meeting has been requested.

**Service Name**: Azure Maps - Azure Maps
**Review Created By**: Will Huang
**Review Date**: 05/23/2024 08:00 AM PT
**Release Plan**: [](https://web.powerapps.com/apps/?release-plan-id=)
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/29153
**Hero Scenarios Link**: [See](https://microsoftapc.sharepoint.com/:p:/t/AzureMapsTW/EaPbd9oBnzBLvP-V-lNQC8IBt2sxScP1TRtIQ_BTANaqkA?e=NqUuoa)
**Core Concepts Doc Link**: [See](https://microsoftapc.sharepoint.com/:p:/t/AzureMapsTW/EaPbd9oBnzBLvP-V-lNQC8IBt2sxScP1TRtIQ_BTANaqkA?e=NqUuoa)

**Description**: SnapToRoads/RouteRange APIs refview

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=b973b121-f208-ef11-9f8a-0022480bd0ae)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by koyasu221b**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

File: specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json [hunk 1]
```diff
@@ -0,0 +1,3597 @@
...
+    },
+    "/route/matrix": {
+      "post": {
+        "description": "**Applies to**: see pricing [tiers](https://aka.ms/AzureMapsPricingTier).\n\n\nUse to get a route matrix showing the travel time and distance for all possible pairs in a list of origins and destinations.\n\n\n\nFor information about routing availability in countries/regions, see [Azure Maps routing coverage](https://learn.microsoft.com/azure/azure-maps/routing-coverage?pivots=route-v2).\n\n>[!Important]\n>By using this feature, you agree to the preview legal terms. See the [Preview Supplemental Terms](https://azure.microsoft.com/en-us/support/legal/preview-supplemental-terms/) for additional details.\n\n",
+        "operationId": "Route_PostRouteMatrix",
+        "consumes": [
+          "application/geo+json"
+        ],
+        "produces": [
+          "application/geo+json",
```
Comment (evaou) [inline specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json:520]: response content type might be "application/geo+json" only

File: specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json [hunk 2]
```diff
@@ -0,0 +1,3597 @@
...
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "operation-location",
+          "final-state-schema": "#/definitions/RouteAsyncOperation"
+        },
+        "consumes": [
+          "application/geo+json"
+        ],
+        "produces": [
+          "application/json"
```
Comment (evaou) [inline specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json:585]: response content type might be "application/geo+json" only

Comment (koyasu221b) [reply specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json:585]: for  async response the result is json.

File: specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json [hunk 3]
```diff
@@ -0,0 +1,3617 @@
...
+        "axleWeight": {
+          "x-ms-client-name": "axleWeightInKilograms",
+          "description": "Weight per axle of the vehicle in kg. A value of 0 means that weight restrictions per axle are not considered.",
+          "format": "int64",
+          "type": "integer",
+          "maximum": 1000000,
+          "minimum": 0,
+          "default": 0
+        },
+        "axleCount": {
```
Comment (FarazGIS) [inline specification/maps/data-plane/Route/preview/2024-06-01-preview/route.json:2547]: @koyasu221b TomTom Route Matrix v2 does not support axleCount. Can we please cross check all params and enums with TomTom's documentation for sync and async?

**Review by JeffreyRichter — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@JeffreyRichter has this been approved by the API review board?

**Comment by JeffreyRichter** [comment]
Yes it has

**Pull request merged**